+config SERIAL_ADM5120
+	bool "ADM5120 serial port support"
+	depends on MIPS_ADM5120
+	select SERIAL_CORE
+	select SERIAL_CORE_CONSOLE
+	help
+	  Driver for the on chip UARTs on the ADM5120 SoC
+
+config ADM5120_NR_UARTS
+	int "Maximum number of ADM5120 serial ports"
+	depends on SERIAL_ADM5120
+	default "2"
+	---help---
+	  Set this to the number of serial ports you want the driver to
+	  support.
+
 config SERIAL_AMBA_PL010
 	tristate "ARM AMBA PL010 serial port support"
 	depends on ARM_AMBA && (BROKEN || !ARCH_VERSATILE)
diff -urN linux-2.6.19.2/drivers/serial/Makefile linux-2.6.19.2-adm5120/drivers/serial/Makefile
--- linux-2.6.19.2/drivers/serial/Makefile	2007-01-10 20:10:37.000000000 +0100
+++ linux-2.6.19.2-adm5120/drivers/serial/Makefile	2007-01-23 01:21:45.000000000 +0100
@@ -20,6 +20,7 @@
 obj-$(CONFIG_SERIAL_8250_HUB6) += 8250_hub6.o
 obj-$(CONFIG_SERIAL_8250_MCA) += 8250_mca.o
 obj-$(CONFIG_SERIAL_8250_AU1X00) += 8250_au1x00.o
+obj-$(CONFIG_SERIAL_ADM5120) += adm5120_uart.o
 obj-$(CONFIG_SERIAL_AMBA_PL010) += amba-pl010.o
 obj-$(CONFIG_SERIAL_AMBA_PL011) += amba-pl011.o
 obj-$(CONFIG_SERIAL_CLPS711X) += clps711x.o
diff -urN linux-2.6.19.2/include/linux/serial_core.h linux-2.6.19.2-adm5120/include/linux/serial_core.h
--- linux-2.6.19.2/include/linux/serial_core.h	2007-01-10 20:10:37.000000000 +0100
+++ linux-2.6.19.2-adm5120/include/linux/serial_core.h	2007-01-23 01:13:39.000000000 +0100
@@ -133,6 +133,9 @@
 #define PORT_S3C2412	73
 
 
+/* ADMtek ADM5120 SoC */
+#define PORT_ADM5120	68
+
 #ifdef __KERNEL__
 
 #include <linux/compiler.h>
